bin/sdl.cc

```diff
@@ -221,14 +220,26 @@ int main(int argc, char** argv)
     ana.do_write_ntuple = result["write_ntuple"].as<int>();
 
     //_______________________________________________________________________________
-    // --cpu
-    if (result.count("cpu"))
+    // check if cpu library was loaded
+    ana.do_run_cpu = false;
+    std::ifstream maps("/proc/self/maps");
+    if (!maps.is_open())
     {
-        ana.do_run_cpu = true;
+        std::cout << "Warning: Failed to open /proc/self/maps. Backend cannot be determined." << std::endl;
     }
     else
     {
-        ana.do_run_cpu = false;
+        std::string line;
+        while (std::getline(maps, line)) {
+            size_t pos = line.find("/"); // Find the starting position of the path
+            if (pos != std::string::npos) {
+                std::string libraryPath = line.substr(pos, line.find(" ", pos) - pos);
+                if (libraryPath.size() >= 14 && libraryPath.compare(libraryPath.size() - 14, 14, "/libsdl_cpu.so") == 0) {
+                    ana.do_run_cpu = true;
+                    break;
+                }
+            }
+        }
```

Comment:
This looks to me like a weird way to determine the backend. Could you explain what is the logic behind it, and what kind of complications led to this?

Reply:
It is definitely a bit of a weird way to do this. I think this was the only way to do this without adding stuff to the library. But maybe we can define a global constant with `#ifdef`s in the library that stores the backend that it was compiled with.